src/bootstrap/tosh.c

```diff
@@ -13,30 +13,74 @@
 #include <stdlib.h>
 #include <ft_printf.h>
 #include <unistd.h>
-#include "../error/error.h"
+
+#include "../env/env.h"
 #include "../input/input.h"
+#include "../lexer/lexer.h"
+#include "../parser/parser.h"
+#include "../error/error.h"
+
+static void		shell_flow_control(const char *const input, t_env *const env)
+{
+	t_list_meta					tokens;
+	struct s_complete_command	*complete_command;
+	bool						extra_input_requested;
+	t_error						err;
+
+	if (lexer_tokenize(&tokens, input) == -1)
+	{
+		ft_dprintf(STDERR_FILENO, "tosh: lexer could not allocate memory\n");
+	}
+	if (tokens.len != 0)
+	{
+		err = parser_parse(&complete_command, &extra_input_requested, &tokens);
+		lexer_clear(&tokens);
+		if (is_error(err))
+		{
+			ft_dprintf(STDERR_FILENO, "tosh: %s\n", err.msg);
+			return ;
+		}
+		if (extra_input_requested)
+		{
+			ft_dprintf(STDERR_FILENO, "extra input requested\n");
+			return ; //not sure how to hack this in yet
+		}
+		// exec_run(complete_command, env);
+		(void)env;
```

Comment:
nit: maybe move the `(void)env;` to the first line of the function? Or remove the `env` parameter?